tests/rptest/services/kgo_verifier_services.py

```diff
@@ -316,19 +316,35 @@ def poll_status(self):
             else:
                 self._stop_requested.wait(self.INTERVAL)
 
+    def join_with_timeout(self):
+        """
+        Join thread with a modest timeout, and raise an exception if
+        we do not succeed.  We expect to join promptly because all our
+        run() is doing is calling to the remote process status endpoint, and
+        that requests.get() has a timeout on it, so should not block.
+
+        This is important because otherwise a stuck join() would hang
+        the entire ducktape test run.
+        """
+        self.join(timeout=10)
+        if self.is_alive():
+            msg = f"Failed to join thread for {self.who_am_i}"
```

Comment:
I'd maybe include the node name in this error, but the association between service name and node is logged on start-up so it's not strictly necessary.